containers/Locale/Provider.js

```diff
@@ -0,0 +1,38 @@
+import { c } from 'ttag';
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useUserSettings, useConfig, useNotifications } from 'react-components';
+import { DEFAULT_TRANSLATION } from 'proton-shared/lib/constants';
+import { loadLocale } from 'proton-shared/lib/i18n';
+
+export const LocaleContext = createContext();
+
+export const LocaleProvider = ({ children }) => {
+    const config = useConfig();
+    const { createNotification } = useNotifications();
+    const [{ Locale = DEFAULT_TRANSLATION } = { Locale }] = useUserSettings();
+    const [state, setState] = useState(Locale);
+
+    // Load the translations when we load the app
+    useEffect(() => {
+        loadLocale(config, Locale).then(() => {
+            setState(Locale); // force refresh children
+        });
+    }, []);
+
+    // Only when the locale changes via event
+    if (state !== Locale) {
+        loadLocale(config, Locale).then(() => {
+            setState(Locale); // force refresh children
+            createNotification({ text: c('Success').t`Locale updated` });
+        });
+    }
+
+    return <LocaleContext.Provider value={state}>{children}</LocaleContext.Provider>;
```

Comment:
Does it really need to provide the current locale?

It can be calculated from settings model right?

Reply:
What do you mean ?

Reply:
Why does it need to provide the locale when it can be retrieved from the settings model?

Reply:
Hum imho it should not be inside Models provider.

Reply:
Why? Every other settings component is reading values from the model

Reply:
because it's the locale